Backend/Backend/Services/EpisodeService.cs

```diff
@@ -54,11 +84,28 @@ public class EpisodeService : IEpisodeService
 
 
         public async Task<bool> DeleteEpisode(Episode episode, DeleteEpisodeRequest deleteEpisodeRequest)
-        {  
-           await _fileService.DeleteFile(episode.AudioFileId.ToString()!);
-           _db.Remove(episode);
-           await _db.SaveChangesAsync();
-           return true;
+        {
+            Guid? guid = Guid.Parse(deleteEpisodeRequest.EpisodeId);
+            Files? file = await _db.File!.FirstOrDefaultAsync(u => u.FileId == episode.AudioFileId);
+
+            if (file != null)
+            {
+               var deleted = _fileService.DeleteFile(file.Path!);
+               if(deleted == true)
+                {
+                _db.Remove(file);
+
+                }
+               else { throw new Exception("Failed to Delete File"); }
+                _db.Remove(episode);
```

Comment:
you should probably delete the episode from the database before removing it from the file system, just in case we have an error half way through, causing a file in the database to exist, but not one in the file system. That situation is easier to handle than this one